fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/BinaryExpression.java

```diff
@@ -23,15 +23,4 @@
  * Interface for all expression that have two children.
  */
 public interface BinaryExpression extends BinaryNode<Expression, Expression, Expression> {
```

Comment:
Do we still need this interface if all the functions declarations are removed?

Reply:
Like `org.apache.doris.nereids.trees.plans.BinaryPlan`, these interfaces are retained for future.

Because these implementations are the same as `BinaryNode`, I remove those due to the redundancy.